pallets/subtensor/tests/swap.rs

```diff
@@ -1048,3 +1048,296 @@ fn test_swap_total_hotkey_coldkey_stakes_this_interval_weight_update() {
+            &new_coldkey
+        ));
+
+        // Verify the swap
+        assert_eq!(Owner::<Test>::get(hotkey), new_coldkey);
+        assert_eq!(TotalColdkeyStake::<Test>::get(new_coldkey), stake_amount);
+        assert!(!TotalColdkeyStake::<Test>::contains_key(old_coldkey));
+        assert_eq!(Stake::<Test>::get(hotkey, new_coldkey), stake_amount);
+        assert!(!Stake::<Test>::contains_key(hotkey, old_coldkey));
+        assert_eq!(OwnedHotkeys::<Test>::get(new_coldkey), vec![hotkey]);
+        assert!(!OwnedHotkeys::<Test>::contains_key(old_coldkey));
+
+        // Verify balance transfer
+        assert_eq!(SubtensorModule::get_coldkey_balance(&new_coldkey), balance);
+        assert_eq!(SubtensorModule::get_coldkey_balance(&old_coldkey), 0);
+
+        // Verify event emission
+        System::assert_last_event(
+            Event::ColdkeySwapped {
+                old_coldkey,
+                new_coldkey,
+            }
+            .into(),
+        );
+    });
+}
+
+#[test]
+fn test_swap_total_coldkey_stake() {
+    new_test_ext(1).execute_with(|| {
+        let old_coldkey = U256::from(1);
+        let new_coldkey = U256::from(2);
+        let stake_amount = 1000u64;
+        let mut weight = Weight::zero();
+
+        // Initialize TotalColdkeyStake for old_coldkey
+        TotalColdkeyStake::<Test>::insert(old_coldkey, stake_amount);
+
+        // Perform the swap
+        SubtensorModule::swap_total_coldkey_stake(&old_coldkey, &new_coldkey, &mut weight);
+
+        // Verify the swap
+        assert_eq!(TotalColdkeyStake::<Test>::get(new_coldkey), stake_amount);
+        assert!(!TotalColdkeyStake::<Test>::contains_key(old_coldkey));
+
+        // Verify weight update
+        let expected_weight = <Test as frame_system::Config>::DbWeight::get().reads_writes(1, 2);
+        assert_eq!(weight, expected_weight);
+    });
+}
+
+#[test]
+fn test_swap_stake_for_coldkey() {
+    new_test_ext(1).execute_with(|| {
+        let old_coldkey = U256::from(1);
+        let new_coldkey = U256::from(2);
+        let hotkey1 = U256::from(3);
+        let hotkey2 = U256::from(4);
+        let stake_amount1 = 1000u64;
+        let stake_amount2 = 2000u64;
+        let mut weight = Weight::zero();
+
+        // Initialize Stake for old_coldkey
+        Stake::<Test>::insert(hotkey1, old_coldkey, stake_amount1);
+        Stake::<Test>::insert(hotkey2, old_coldkey, stake_amount2);
+
+        // Populate OwnedHotkeys map
+        OwnedHotkeys::<Test>::insert(old_coldkey, vec![hotkey1, hotkey2]);
+
+        // Perform the swap
+        SubtensorModule::swap_stake_for_coldkey(&old_coldkey, &new_coldkey, &mut weight);
+
+        // Verify the swap
+        assert_eq!(Stake::<Test>::get(hotkey1, new_coldkey), stake_amount1);
+        assert_eq!(Stake::<Test>::get(hotkey2, new_coldkey), stake_amount2);
+        assert!(!Stake::<Test>::contains_key(hotkey1, old_coldkey));
+        assert!(!Stake::<Test>::contains_key(hotkey2, old_coldkey));
+
```

Comment:
```suggestion
assert_eq!(TotalHotkeyStake::<Test>::get(hotkey1), stake_amount1);
assert_eq!(TotalHotkeyStake::<Test>::get(hotkey2), stake_amount2);
assert_eq!(TotalStake::<Test>::get(), stake_amount1 + stake_amount2);
assert_eq!(TotalIssuance::<Test>::get(), stake_amount1 + stake_amount2);
```
Would be good to check `TotalHotkeyStake`,  `TotalStake` and `TotalIssuance` doesn't change after swap.